Scanner panics when GitHub rules and tokens are empty

## Summary

The scanner crashes and enters a restart loop when the GitHub scan phase runs without any configured GitHub rules or tokens. This prevents later scan providers, including Postman, from running.

## Environment

- GShark: `v2.1.7`
- Deployment: Docker Compose
- Scanner command: `./gshark scan`
- Architecture: `linux/amd64`

## Steps to reproduce

1. Initialize GShark without adding a GitHub token.
2. Leave GitHub rules empty.
3. Start the scanner container with `./gshark scan`.
4. Observe the scanner after it reaches the GitHub phase.

## Actual behavior

The scanner reports zero GitHub rules, then panics while indexing the empty token slice:

```text
There is no client for Gitlab, please check if you specify Gitlab token
[GShark]2026/07/29-13:59:23 info Complete the scan of GitLab, start to sleep 900 seconds
[GShark]2026/07/29-13:59:23 info Rules of search code is empty
Github fetch 0 rules, begin the scan task
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/madneal/gshark/search/githubsearch.GetGithubClient()
    /go/src/github.com/madneal/gshark/server/search/githubsearch/gitclient.go:42 +0x10b
github.com/madneal/gshark/search/githubsearch.Search(...)
    /go/src/github.com/madneal/gshark/server/search/githubsearch/search.go:22
github.com/madneal/gshark/search/githubsearch.RunTask(...)
    /go/src/github.com/madneal/gshark/server/search/githubsearch/search.go:114
github.com/madneal/gshark/search.ScanTask()
    /go/src/github.com/madneal/gshark/server/search/scan.go:22
```

With a restart policy enabled, the container repeatedly restarts and never reaches the Postman scan phase.

## Root cause

`githubsearch.RunTask` calls `Search(rules)` even when `rules` is empty. `Search` immediately calls `GetGithubClient`, and `GetGithubClient` accesses `tokens[0]` without checking whether `ListTokenByType("github")` returned an empty slice.

## Expected behavior

- Skip the GitHub phase when no enabled GitHub rules exist.
- Return a descriptive error when GitHub rules exist but no GitHub token is configured.
- Continue to subsequent scan providers instead of terminating the scanner process.

## Suggested fix

1. Return early from `Search` when `len(rules) == 0`.
2. Check `len(tokens)` before accessing `tokens[0]` in `GetGithubClient`.
3. Add a regression test covering an empty GitHub rule list.

--- a/server/search/githubsearch/gitclient.go
+++ b/server/search/githubsearch/gitclient.go
@@ -39,6 +39,9 @@ func GetGithubClient() (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	if len(tokens) == 0 {
+		return nil, errors.New("github client initialization failed: no token configured")
+	}
 	client := InitClient(tokens[0].Content)
 	if client == nil {
 		err = errors.New("github Client initial failed, please add token")

--- a/server/search/githubsearch/search.go
+++ b/server/search/githubsearch/search.go
@@ -19,6 +19,9 @@ import (
 )
 
 func Search(rules []model.Rule) {
+	if len(rules) == 0 {
+		return
+	}
 	client, err := GetGithubClient()
 	if err != nil {
 		global.GVA_LOG.Error("GetGithubClient err", zap.Error(err))

--- a/server/search/githubsearch/search_test.go
+++ b/server/search/githubsearch/search_test.go
@@ -20,3 +20,7 @@ func TestSearch(t *testing.T) {
 	})
 	Search(rules)
 }
+
+func TestSearchWithNoRules(t *testing.T) {
+	Search(nil)
+}

